test/index.test.js

```diff
@@ -1,9 +1,326 @@
 'use strict';
 
+const Hapi = require('hapi');
 const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
 
-describe('index test', () => {
-    it('fails', () => {
-        assert.isTrue(false);
+sinon.assert.expose(assert, { prefix: '' });
+
+/**
+ * helper to generate a nodemailer mock
+ * @method getUserMock
+ * @return {Function}         Stubbed function
+ */
+function getNodemailerMock() {
+    const sendMailMock = {
+        sendMail: sinon.stub().yieldsAsync()
+    };
+
+    return { createTransport: sinon.stub().returns(sendMailMock) };
+}
+
+describe('index', () => {
+    let serverMock;
+    let configMock;
+    let notifier;
+    let buildDataMock;
+    let nodemailerMock;
+    let EmailNotifier;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+
+        nodemailerMock = getNodemailerMock();
+        mockery.registerMock('nodemailer', nodemailerMock);
+
+        // eslint-disable-next-line global-require
+        EmailNotifier = require('../index');
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    afterEach(() => {
```

Comment:
Do you even need this afterEach? I figured this would actually break tests due to it deregistering your mocks after every test, and you are now not setting up mocks before each test.